extensions/commands/cci/cmd_export_all_versions.py

```diff
@@ -0,0 +1,130 @@
+    group.add_argument('-p', '--path', action=OnceArgument, help="Path of the global recipe folder")
+    args = parser.parse_args(*args)
+
+    if args.name is not None:
+        recipes_to_export = [args.name]
+    elif args.list is not None:
+        recipes_to_export = yaml.safe_load(open(args.list, 'r'))["recipes"]
+    else:
+        recipes_to_export = os.listdir(args.path)
+
+    out = ConanOutput()
+
+    # Result output variables, these should always be returned
+    exported = {}
+    exported_refs = []
+    exported_with_revision = []
+    failed = dict()
+
+    for item in recipes_to_export:
+        recipe_name = item if not isinstance(item, dict) else list(item.keys())[0]
+        folders_to_export = item[recipe_name][0]['folders'] if isinstance(item, dict) else None
+        out.verbose(f"Processing recipe '{recipe_name}'")
+
+        recipe_folder = os.path.join("recipes", recipe_name)
+        if not os.path.isdir(recipe_folder):
+            raise ConanException(f"Invalid user input: '{recipe_name}' folder does not exist")
+
+        config_file = os.path.join(recipe_folder, "config.yml")
+        if not os.path.isfile(config_file):
+            raise ConanException(f"The file {config_file} does not exist")
+
+        config = yaml.safe_load(open(config_file, "r"))
+        for version in config["versions"]:
+            recipe_subfolder = config["versions"][version]["folder"]
+            if folders_to_export and recipe_subfolder not in folders_to_export:
+                continue
+            conanfile = os.path.join(recipe_folder, recipe_subfolder, "conanfile.py")
+            if not os.path.isfile(conanfile):
+                raise ConanException(f"The file {conanfile} does not exist")
+
+            out.verbose(f"Exporting {recipe_name}/{version} from {recipe_subfolder}/")
+            try:
+                ref = conan_api.export.export(os.path.abspath(conanfile), recipe_name, version, None, None)
+                out.verbose(f"Exported {ref}")
+
+                if recipe_name not in exported:
+                    exported[recipe_name] = []
+                exported[recipe_name].append(ref)
+                exported_refs.append(ref)
+
+                exported_with_revision.append(f"{ref[0].name}/{ref[0].version}#{ref[0].revision}")
+            except Exception as e:
+                failed.update({f"{recipe_name}/{recipe_subfolder}": str(e)})
+
+    out.title("EXPORTED RECIPES")
+    for item in exported.keys():
+        out.info(f"{item}: exported {len(exported[item])} versions")
+
+    out.title("FAILED TO EXPORT")
+    for key, value in failed.items():
+        out.info(f"{key}, {value}")
+
+    out.title("RECIPE REFERENCES")
+    for item in exported_with_revision:
+        out.info(f"{item}")
+
+    out.title("REFERENCE LIST")
+
+    versions_list = [f"{item[1]}" for item in exported_refs]
+    out.info(f'{versions_list}')
```

Comment:
Maybe add some comments?

it's confusing why you'd print `name/version#revision` in the section above and then again dump the list of versions (with no recipe name)

I dont know why this is helpful in the CLI stderr but no included in the JSON output 🤔

Reply:
This was needed by @jcar87 to generate a list to then pass to tapaholes, but I admit that outside of that, that output might not be the most useful one. I've moved the output to its own text formatter for now